### PR TITLE
Кузнецов Виктор - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by just-sparta**

### CODE REVIEW

**Comment by just-sparta** [comment]
Please review )
@kapkaevn @ZinovievVladimir @Fulmo

**Comment by Fulmo** [comment]
👍🏻

File: code/kuznetsov_victor_lab1/debt.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class DebtExpenses(object):
+    def __init__(self, req_sum: int, percent_rate: float, period: int):
+        if req_sum < 0 or percent_rate < 0 or period < 0:
+            raise ValueError('Values must be positive numbers!')
```
Comment (vzinovie) [inline code/kuznetsov_victor_lab1/debt.py:4]: `positive` or `non-negative`?

File: code/kuznetsov_victor_lab1/debt.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        if not isinstance(req_sum, int) or not isinstance(period, int) or not isinstance(percent_rate, float):
+            raise TypeError('Req. sum and period must be integer!')
+        self.req_sum = req_sum
+        self.percent_rate = percent_rate
+        self.period = period
+
+    def equal_amounts_repayment(self, year: int):
+        if year < 1 or year > self.period:
+            raise Exception('Year must be less than {}!'.format(self.period))
```
Comment (vzinovie) [inline code/kuznetsov_victor_lab1/debt.py:13]: What about year < 1 ?

File: code/kuznetsov_victor_lab1/debt.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        expenses = self.req_sum / self.period
+        percent_for_year = (self.req_sum - expenses * (year - 1)) * self.percent_rate
+        payment = percent_for_year + expenses
+        return payment, expenses
+
+    def equal_payments_repayment(self, year: int):
+        if year < 1 or year > self.period:
+            raise Exception('Year must be less than {}!'.format(self.period))
```
Comment (vzinovie) [inline code/kuznetsov_victor_lab1/debt.py:22]: Same

File: code/kuznetsov_victor_lab1/debt.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+        convertion_rate = (1 - (1 + self.percent_rate) ** (-self.period)) / self.percent_rate
+        payment = self.req_sum / convertion_rate
+
+        i = 0
+        expenses = 0
+        remaining = self.req_sum
+        while i < year:
```
Comment (vzinovie) [inline code/kuznetsov_victor_lab1/debt.py:30]: I think `for i in range(year):` will be better here.

**Review by vzinovie — CHANGES_REQUESTED** [review]

**Comment by vzinovie** [comment]
👍

**Review by sawirani — APPROVED** [review]

**Review by sawirani — APPROVED** [review]

**Pull request merged**